dataeng/jobs/AnalyticsEmailOptinWorker.groovy

```diff
@@ -0,0 +1,49 @@
+job ('analytics-email-optin-worker-dsl-test') {
+
+    parameters {
+        stringParam('NOTIFICATION_EMAILS')
+        stringParam('MASTER_WORKSPACE')
+        stringParam('ORG_CONFIG_PATH')
+        stringParam('GPG_KEYS_PATH')
+        stringParam('DATE')
+        stringParam('CONFIG_PATH')
+        stringParam('OUTPUT_BUCKET')
+        stringParam('EXPORTER_VENV')
+        stringParam('ORG')
+        stringParam('OUTPUT_PREFIX')
+        stringParam('PLATFORM_VENV')
+    }
+
+    throttleConcurrentBuilds {
+        maxPerNode(5)
+        maxTotal(5)
+    }
+
+    concurrentBuild()
+
+    multiscm {
+        git {
+            remote {
+                url('$BAKED_CONFIG_SECURE_REPO')
+                branch('*/master')
+                credentials('1')
+            }
+            extensions {
+                relativeTargetDirectory('config/baked-config-secure')
+            }
+        }
+    }
+
+    wrappers {
+        timestamps()
+        buildName('#${BUILD_NUMBER} ${ENV,var="ORG"}')
+    }
+
+    steps {
+        shell(readFileFromWorkspace("dataeng/resources/email-optin-worker.sh"))
+    }
+
+    publishers {
+        textFinder("Task OrgEmailOptInTask failed fatally", '', true, false, false)
```

Comment:
Do we want to mark this build as unstable if the text is matched ?

Reply:
No I think when it actually fails fatally then the master should be marked as failure.  I think unstable doesn't happen on the Email Optin job.  The output for that job is pretty stable.  I think unstable tends to happen when there are partially built courses being exported during the rest of the exporter process.